src/tree.service.ts

```diff
@@ -24,6 +25,8 @@ export class TreeService {
   public nodeExpanded$: Subject<NodeExpandedEvent> = new Subject<NodeExpandedEvent>();
   public nodeCollapsed$: Subject<NodeCollapsedEvent> = new Subject<NodeCollapsedEvent>();
 
+  private controllers: Map<string | number, TreeController> = new Map();
```

Comment:
I've just found a problem in my implementation: there is now way to set a new node's id, and this will make it impossible to get it's controller. I will think about a solutions of this issue.